api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1850,6 +1851,43 @@ type AzurePlatformSpec struct {
 	// +immutable
 	// +required
 	SecurityGroupID string `json:"securityGroupID,omitempty"`
+
+	// MSIClientIDs contains the client IDs related to the managed identities needed for the following control plane
+	// components: azure cloud provider.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	MSIClientIDs *ControlPlaneManagedServiceIdentities `json:"msiClientIDs,omitempty"`
+}
+
+type ControlPlaneManagedServiceIdentities struct {
+	// azureCloudProviderMSIClientID is the client ID of a pre-existing managed identity ID of that will be associated
+	// with the azure cloud provider, aka ccm. The managed identity is expected to be in
+	// AzurePlatformSpec.ResourceGroupName.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	AzureCloudProviderMSIClientID string `json:"azureCloudProviderMSIClientID"`
+
+	// clusterAPIAzureMSIClientID is the client ID of a pre-existing managed identity ID of that will be associated
+	// with cluster-api azure. The managed identity is expected to be in AzurePlatformSpec.ResourceGroupName.
```

Comment:
> The managed identity is expected to be in AzurePlatformSpec.ResourceGroupName.

I think we need to make this more flexible.  The `AzurePlatformSpec.ResourceGroupName.` value that we set to in the HostedCluster from CS is the Managed Resource Group Name. The end-user will precreate the Azure Managed Identities which means that they will *always* be outside of the managed resource group.

This comment also applies to all the other attributes where this restriction has been documented.

The same approach should be followed for the data plane identities too (which I understand is being worked on in another PR).

cc @bennerv

Reply:
Yeah, Ben and I have discussed they would be in a different resource group. The API will be updated before this PR is ready for review.

Further work on this PR is stalled until we receive the sidecar container.

Reply:
:+1: thanks